iothub/device/src/PnPOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal class PnPOptions
+    {
+        internal string ModelId { get; set; } = default;
```

Comment:
This is not added as a simple string ModelId, but as a defined type because the pipeline context where it gets added to is a dictionary <type, value>.